src/torchcodec/samplers/_implem.py

```diff
@@ -0,0 +1,138 @@
+import random
+from typing import List, Optional
+
+import torch
+
+from torchcodec.decoders import FrameBatch, SimpleVideoDecoder
+
+
```

Comment:
Below I factored-out some validation logic into functions. This isn't strictly necessary for this PR, but we'll need the exact same logic for the `clips_at_regular_indices()` sampler. So I prefer extracting the logic now, so that we can minimize the diff changes of the `clips_at_regular_indices()` PR.